### ISSUE TITLE
Training duration/batch size

### ISSUE THREAD

**Issue by josch14** [ttengwang/dense-video-captioning-pytorch#5]
Hi. First of all, thanks for publishing your implementation code!

I'm quite unexperienced in training dense video captioining models and therefore I have some questions. 

1) You mention to first train the model with tsrm_cmg_hrnn.yml, and afterwards with tsrm_cmg_hrnn_RL_enlarged_trainset.yml. Was this also the way you did it for the model that you submitted for the challenge? If not, how you did it?

2) Since both mentioned models contain a hrnn, I'm only allowed to train them with a batch size of 1, right? Meaning I will just use a tiny amount of my gpu which should not be desirable?
However, one epoch is done quite fast in a matter of some minutes. Compard to the [model](https://github.com/salesforce/densecap) u linked this is much quicker, where one epoch takes around 20 to 30 minutes for me. Why is that? 

Generally, I really would like to know for which batch size which training duration / how many epochs I can expect to do.

ps: These questions shouldn't be an issue but rather a personal mail to you, right? :)

**Comment by ttengwang** [comment]
1. Yes. 
2. **Batch size and gpu**: Yes, batch size = 1 and a single GPU are enough to train a good model. 
For tsrm_cmg_hrnn.yml, I recommend setting the max epoch number to 25.
For tsrm_cmg_hrnn_RL_enlarged_trainset.yml, the larger epochs and larger train_proposal_sample_num, the better. 
**Why quicker than Masked Transformer**: the reason might be 1) we only train the captioning module, while MT train the captioning module and proposal generation together. 2) we only feed ground-truth proposals (~3.6 on average) for training, while MT uses a larger number of proposals. 3) During decoding, Transformer has a larger computation than RNN

**Comment by josch14** [comment]
Thanks!
